libraries/redcore/api/hal/hal.php

```diff
@@ -628,6 +628,26 @@ public function apiCreate()
 		$functionName = RApiHalHelper::attributeToString($this->operationConfiguration, 'functionName', 'save');
 		$data = $this->triggerFunction('processPostData', $this->options->get('data', array()), $this->operationConfiguration);
 
+		// Remove primary keys for "model" and "table" data mode
+		$dataMode = RApiHalHelper::attributeToString($this->operationConfiguration, 'dataMode');
+
+		if (($dataMode == 'model') || ($dataMode == 'table'))
+		{
+			// Get the primary keys
+			$primaryKeys = $model->getTable()->getKeyName(true);
```

Comment:
I don't think I can use this function. Because if we use this function, we must set the primary key in XML file.